### PR TITLE
Using the new RNN module for speech demo

### PR DESCRIPTION
**Pull request by yzhang87**
1) Fix a typo in executor_group.
2) Add one more parameter to customize metric: can modify both metric and the number of samples per batch.
3) Using the new RNN module
4) optimize frame accuracy (53% -> 63%)
5) Fix inconsistence of comments and actual code in optimizer (learning_rate) for Adam optimizer.

### CODE REVIEW

**Comment by yzhang87** [comment]
@pluskid most of the change was only on the speech-demo, except the change for metric.py
@piiswrong , I found that after the refactor of optimizer.py, the comments and actual code are inconsistent. I only fixed for Adam. Would you mind check it and also all the other solvers? Right now, it grab the learning rate from base class which is too huge for some optimizer, e.g. Adam.

**Comment by pluskid** [comment]
yu is referring to the default value for the learning rate that was originally specified for different optimizers. they are recently refactored to inherit a single uniform default value, which might not be good  for all optimizers at the same time. i guess it might be good to consider bringing them back.

**Comment by pluskid** [comment]
this looks good to me.

**Comment by pluskid** [comment]
BTW: @yzhang87 after #1938 is merged, we can then set `use_ignore=True` and `ignore_label=0` to ignore the padded examples in bucketing. Make sure to distinguish between padding and silence in this case.

**Comment by pluskid** [comment]
@yzhang87 Can you rebase again? Let's try to get this merged tonight.

**Comment by yzhang87** [comment]
done.

On Fri, Apr 22, 2016 at 10:27 PM, Chiyuan Zhang <notifications@github.com>
wrote:

> @yzhang87 <https://github.com/yzhang87> Can you rebase again? Let's try
> to get this merged tonight.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/dmlc/mxnet/pull/1930#issuecomment-213652193>
>



-- 
Yu Zhang
Computer Science and Artificial Intelligence Laboratory (CSAIL)
Massachusetts Institute of Technology
